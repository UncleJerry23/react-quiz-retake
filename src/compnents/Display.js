import React from 'react';
import PropTypes from 'prop-types';

function Display({ meme }) {
  return (
    <>
      <p>{meme.headerText}</p>
      <img src={meme.image} />
      <p>{meme.footerText}</p>
    </>
  );
}

Display.propTypes = {
  meme: PropTypes.shape({
    headerText: PropTypes.string.isRequired,
    footerText: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  }).isRequired
};

export default Display;
