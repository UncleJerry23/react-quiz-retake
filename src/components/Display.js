import React from 'react';
import PropTypes from 'prop-types';

function Display({ meme }) {

  const imgStyle = {
    width: '100px'
  };

  return (
    <>
      <p>{meme.headerText}</p>
      <img src={meme.image} style={imgStyle}/>
      <p>{meme.footerText}</p>
    </>
  );
}

Display.propTypes = {
  meme: PropTypes.shape({
    headerText: PropTypes.string.isRequired,
    footerText: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  }).isRequired
};

export default Display;
