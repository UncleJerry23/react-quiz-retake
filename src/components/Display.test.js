import React from 'react';
import { shallow } from 'enzyme';
import Display from './Display';

describe('Display component', () => {
  it('renders a Display', () => {
    const meme = {
      headerText: 'meme baby meme',
      footerText: 'aint it neat',
      image: 'asdfg.com.png'
    };

    const wrapper = shallow(<Display meme={meme} />);
    expect(wrapper).toMatchSnapshot();
  });
});
