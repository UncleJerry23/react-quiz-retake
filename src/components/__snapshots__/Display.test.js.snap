// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component renders a Display 1`] = `
<Fragment>
  <p>
    meme baby meme
  </p>
  <img
    src="asdfg.com.png"
    style={
      Object {
        "width": "100px",
      }
    }
  />
  <p>
    aint it neat
  </p>
</Fragment>
`;
